cg/cg_finch.jl

```diff
@@ -25,60 +25,53 @@ function ssymv_finch(y, A, x)
     y
 end
 
+# Using algorithm from https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/e5d10e22552916e7dbd73d0a9ece02d873dee375/src/cg.jl
 function cg_finch_kernel(x, A, b, l)
     (n, m) = size(A)
     @assert n == m
 
-    r = Fiber!(Dense(Element(0.0), n))
-    _r = Fiber!(Dense(Element(0.0), n))
-    p = Fiber!(Dense(Element(0.0), n))
-    _p = Fiber!(Dense(Element(0.0), n))
-    _x = Fiber!(Dense(Element(0.0), n))
-    Ap = Fiber!(Dense(Element(0.0), n))
+    _x = Tensor(Dense(Element(0.0)), zeros(n))
```

Comment:
instead of using `Tensor(Dense(Element(0.0)), zeros(n))`, use `Tensor(Dense(Element(0.0)), undef, n)`